### DISCUSSION TITLE
Litestream environment variables

### DISCUSSION THREAD

**Discussion by danielmichaels** [robherley/snips.sh#51] [Ideas]
Thanks for creating snips - it's awesome.

Is there a way to run Litestream without the docker-compose file but instead just use some environment variables?

Background:

I'm running my own snips.sh on a [CapRover](https://caprover.com) instance. CapRover doesn't really support `docker-compose` without an (IMO) [janky workaround](https://caprover.com/docs/docker-compose.html). 

It does have great support for populating environment variables.

I do this in one of my own projects, where I have a `litestream.yml` included and environment variables entirely populate it; [example](https://github.com/danielmichaels/tars.run/blob/main/litestream.yml).

What do you think? Something worth exploring?

**Comment by robherley** [comment]
> Thanks for creating snips - it's awesome.

👋 Thanks for using it!

You can totally use litestream without docker-compose, they have [great documentation](https://litestream.io/guides/) on how to do so. 

I'm not familiar with CapRover, but if you need everything to be within a single dockerfile/container, you can use setup a [process manager](https://docs.docker.com/config/containers/multi-service_container/#use-a-process-manager) like `supervisord` to run snips and litestream in a single container. e.g. start the Dockerfile with `FROM ghcr.io/robherley/snips.sh...` install litestream in the dockerfile and make supervisord point to both `/usr/bin/snips.sh` and the litestream binary.

Otherwise, you can use litestream outside of that container and point it to the sqlite database file 👍
